wgengine/monitor: close winMin.closeHandle

alexbrainman opened the pull request:
eccc167 introduced winMin.closeHandle. The commit opens the handle,
but it never closes it.

Windows handles are not boundless, and should be closed.

Updates #921

Signed-off-by: Alex Brainman <alex.brainman@gmail.com>

bradfitz commented on wgengine/monitor/monitor_windows.go:
```diff
@@ -87,6 +87,7 @@ func (m *winMon) Close() error {
 	m.cancel()
 	m.pollTicker.Stop()
 	windows.SetEvent(m.closeHandle) // wakes up any reader blocked in Receive
+	windows.CloseHandle(m.closeHandle)
```
Do we need to wait for the goroutine blocked in `WaitMultipleObjects` (or not-yet-blocked in it) to return before closing the handle?

alexbrainman replied:
We do indeed. I did not consider scenario where m.closeHandle can be used after it was closed. Fixed now.

I did not want to introduce extra complexity. But I don't see other way.

alexbrainman commented:
@bradfitz do you still want this change?

I think we should submit it. I don't have any alternative version yet.

Alex

bradfitz commented:
I fixed a typo and updated the commit message to remove inaccurate mentions of `winMin`, which isn't involved here.

alexbrainman commented:
> I fixed a typo and updated the commit message to remove inaccurate mentions of `winMin`, which isn't involved here.

Sounds good. Thank you.

It was meant to be `winMon.closeHandle`, not `winMin.closeHandle`. But `closeHandle` is good enough too.

Alex
